### PR TITLE
Fixes #37477 - Make host links be settings aware

### PR DESCRIPTION
**Pull request by ofedoren**

### CODE REVIEW

File: webpack/components/OpenscapRemediationWizard/steps/ReviewHosts.js [hunk 1]
```diff
@@ -161,8 +162,8 @@ const ReviewHosts = () => {
   const columns = {
     name: {
       title: __('Name'),
-      wrapper: ({ id, display_name: displayName }) => (
-        <a href={foremanUrl(`hosts/${id}`)}>{displayName}</a>
+      wrapper: ({ name, display_name: displayName }) => (
+        <a href={foremanUrl(`${getHostsPageUrl(true)}/${name}`)}>{displayName}</a>
```
Comment (MariaAga) [inline webpack/components/OpenscapRemediationWizard/steps/ReviewHosts.js:166]: I suggested the same mistake I did in yesterdays PR :upside_down_face: 
I meant `HOSTS_PATH` from `import { HOSTS_PATH } from 'foremanReact/routes/Hosts/constants';`
HOSTS_PATH is for `/new/hosts` , for single hosts 
getHostsPageUrl is for the experimental index host page

Comment (ofedoren) [reply webpack/components/OpenscapRemediationWizard/steps/ReviewHosts.js:166]: But why is that a mistake? The link is built right and we do use it in other places as well: 

https://github.com/theforeman/foreman_openscap/blob/master/webpack/components/OpenscapRemediationWizard/ViewSelectedHostsLink.js#L17

https://github.com/theforeman/foreman_openscap/blob/master/webpack/components/OpenscapRemediationWizard/steps/ReviewRemediation.js#L123

Comment (MariaAga) [reply webpack/components/OpenscapRemediationWizard/steps/ReviewHosts.js:166]: I dont think `getHostsPageUrl` is supposed to be used this way anyway 
```
export const getHostsPageUrl = displayNewHostsPage =>
  displayNewHostsPage ? '/new/hosts' : '/hosts';

export const useForemanHostsPageUrl = () => {
  const { displayNewHostsPage } = useForemanSettings();
  return getHostsPageUrl(displayNewHostsPage);
};
```
`useForemanHostsPageUrl` is using the setting `displayNewHostsPage` which is for the index page, and not for the show page

Comment (MariaAga) [reply webpack/components/OpenscapRemediationWizard/steps/ReviewHosts.js:166]: Created https://github.com/theforeman/foreman/pull/10174 so foreman context will use the settings for both index and show page for host url

**Comment by ofedoren** [comment]
Thanks, @MariaAga, I've updated all the links in the wizard to use helpers from core.

File: webpack/components/OpenscapRemediationWizard/steps/ReviewHosts.js [hunk 2]
```diff
@@ -161,8 +163,15 @@ const ReviewHosts = () => {
...
     name: {
       title: __('Name'),
-      wrapper: ({ id, display_name: displayName }) => (
-        <a href={foremanUrl(`hosts/${id}`)}>{displayName}</a>
+      wrapper: ({ name, display_name: displayName }) => (
+        <Button
+          component="a"
+          variant="link"
+          target="_blank"
+          href={foremanUrl(`${useForemanHostDetailsPageUrl()}${name}`)} // eslint-disable-line react-hooks/rules-of-hooks
```
Comment (MariaAga) [inline webpack/components/OpenscapRemediationWizard/steps/ReviewHosts.js:171]: Please move `useForemanHostDetailsPageUrl` out so it wont break react by accident

Comment (ofedoren) [reply webpack/components/OpenscapRemediationWizard/steps/ReviewHosts.js]: It's really... weird that this could break anything :/

Updated.

**Review by MariaAga — APPROVED** [review]
code looks good but I didnt test it :+1:

**Pull request merged**